### PR TITLE
Update protobuf from 3.0.0-beta-1 to 3.13.0.0

### PR DESCRIPTION
**Pull request by cameronfrandsen**
Currently maintaining namespace xpro(even though protobuf now has a namespace protobuf)
Current we changing the install step to not put down symlinks.
I have removed SYSTEM_LIBRARIES as it builds on centos and ubuntu
removing .pc files

I know we had talked about not making this pull till https://github.com/smanders/protobuf/pull/2 gets merged. but I was thinking about it earlier today, its hard to test that pull without this pull. So I figured I could make this pull that way its easier for you to see how it builds now. Then after the other pull gets merged, I will make the correct changes here.

### CODE REVIEW

File: patches/protobuf.patch [hunk 1]
```diff
@@ -1,28 +1,30 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
 new file mode 100644
-index 0000000..f74088f
+index 000000000..d73089c43
```
Comment (cameronfrandsen) [inline patches/protobuf.patch:3]: I'm not really sure what to do with this, as my git was a little more updated....

File: projects/protobuf.cmake [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # protobuf
 xpProOption(protobuf DBG)
-set(VER 3.0.0-beta-1)
+set(VER 3.13.0)
 set(REPO github.com/protocolbuffers/protobuf)
-set(FORK github.com/smanders/protobuf)
+set(FORK github.com/cameronfrandsen/protobuf)
```
Comment (cameronfrandsen) [inline projects/protobuf.cmake:5]: This is something that will definitely have to change when the other pull is merged. I will revert it back to smanders

File: patches/protobuf.patch (hunk 1 above)
Comment (smanders) [reply patches/protobuf.patch:3]: this is a common thing as old patch files get updated

File: projects/protobuf.cmake (hunk 2 above)
Comment (smanders) [reply projects/protobuf.cmake:5]: technically you can make the patch without changing this line... just make a remote in the cloned repo and make the patch -- but this is fine, I can deal with this

File: projects/protobuf.cmake [hunk 3]
```diff
@@ -32,11 +32,12 @@ function(build_protobuf)
...
     )
   set(XP_CONFIGURE
-    -DBUILD_TESTING=OFF # we don't have gmock, unless we switch to a release tar ball
+    -Dprotobuf_BUILD_TESTS=OFF # we don't have gmock, unless we switch to a release tar ball
     -DZLIB_MODULE_PATH=ON # with this option ON, we don't need -DZLIB=ON
     -DCMAKE_INSTALL_LIBDIR=lib # without this *some* platforms (RHEL, but not Ubuntu) install to lib64
     -DXP_NAMESPACE:STRING=xpro
-    -DPROTOBUF_VER=${VER}
+    -DPROTOBUF_VER=${VER}.0
+    -DCMAKE_INSTALL_CMAKEDIR=lib/cmake/protobuf_${VER}
```
Comment (smanders) [inline projects/protobuf.cmake:40]: this answers one of my questions in the protobuf review... thanks!

File: projects/use/usexp-protobuf-config.cmake [hunk 4]
```diff
@@ -13,6 +13,7 @@ set(${PRJ}_VER "@VER@ [@PROJECT_NAME@]")
 set(nsPrefix xpro::) # TRICKY: nsPrefix also used in protobuf-module.cmake
 set(FPHSA_NAME_MISMATCHED TRUE) # find_package_handle_standard_args NAME_MISMATCHED (prefix usexp-)
 # protobuf installs a config file which includes -targets.cmake and -module.cmake
+set(protobuf_MODULE_COMPATIBLE ON) # necessary for GENERATE_PROTOBUF_CPP
```
Comment (smanders) [inline projects/use/usexp-protobuf-config.cmake:16]: I'll probably move this line... the comment above it applies to the line below

**Review by smanders — APPROVED** [review]
thanks for all your work on this!

**Pull request merged**